crates/sui-core/src/authority.rs

```diff
@@ -1825,8 +1826,13 @@ impl AuthorityState {
             .compute_object_reference())
     }
 
-    // TODO: Audit every call to this function to make sure there are no data races during reconfig.
-    pub fn get_sui_system_state_object(&self) -> SuiResult<SuiSystemState> {
+    /// This function should be called one and exactly once during reconfiguration.
+    pub fn get_sui_system_state_object_during_reconfig(&self) -> SuiResult<SuiSystemState> {
+        self.database.get_sui_system_state_object()
```

Comment:
is it possible to assert that `execution_lock` is held for write when this is called?

Reply:
i guess we don't hold it yet when this is called...